File: Missing_Number_1.cpp
/*
Link of the question : https://leetcode.com/problems/missing-number/
Leetcode question number : 268
*/
class Solution
{
public:
    int missingNumber(vector<int> &nums)
    {
        int n = 0;
        int result = nums.size();
        for (auto x : nums)
        {
            result = result ^ x;
            result = result ^ n;
            n++;
        }
        return result;
    }
};

File: Missing_Number_1.py
'''
Link of the question : https://leetcode.com/problems/missing-number/
Leetcode question number : 268
'''


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        totalf = n*(n+1)//2
        totals = sum(nums)
        return totalf-totals
